File: 05-05-2026/narcissistic_number.py
def is_narcissistic(n):

    return n == sum([int(x) ** len(str(n)) for x in str(n)])


print(is_narcissistic(153)); # True
print(is_narcissistic(154)); # False
print(is_narcissistic(371)); # True
print(is_narcissistic(512)); # False
print(is_narcissistic(9)); # True
print(is_narcissistic(11)); # False
print(is_narcissistic(9474)); # True
print(is_narcissistic(6549)); # False

File: 05-05-2026/narcissistic_number.js
function isNarcissistic(n) {
  return (
    n
      .toString()
      .split("")
      .reduce(
        (acc, digit) => acc + Math.pow(parseInt(digit), n.toString().length),
        0,
      ) === parseInt(n)
  );
}

console.log(isNarcissistic(153)); // true
console.log(isNarcissistic(154)); // false
console.log(isNarcissistic(371)); // true
console.log(isNarcissistic(512)); // false
console.log(isNarcissistic(9)); // true
console.log(isNarcissistic(11)); // false
console.log(isNarcissistic(9474)); // true
console.log(isNarcissistic(6549)); // false
